python_benchmarks/core.py

```diff
@@ -0,0 +1,109 @@
+import torch
+from torch.profiler import profile, ProfilerActivity
+
+gpu_name = torch.cuda.get_device_name(0)
+if "A100" in gpu_name:
+    GPU = "A100"
+elif "H100" in gpu_name:
+    GPU = "H100"
+elif "V100" in gpu_name:
+    GPU = "V100"
+else:
+    raise ValueError("Invalid GPU name")
+
+# set different properties for different GPUs
+if GPU == "A100":
+    L2_CACHE_SIZE = 40 * 1024 * 1024
```

Comment:
We should get these hardware parameters programmatically. Is there an interface to access the device properties from Python?

Reply:
`torch.cuda` has SM count but is missing L2 cache size, as opposed to `at::cuda` in CPP. Let me see if we can fetch this some other way.